kerastuner/engine/oracle.py

```diff
@@ -17,90 +17,244 @@
+    def create_trial(self, tuner_id):
+        # Allow for multi-worker DistributionStrategy within a Trial.
+        if tuner_id in self.ongoing_trials:
+            return self.ongoing_trials[tuner_id]
 
-        If not overridden, this method does nothing.
+        trial_id = trial_lib.generate_trial_id()
 
-        Args:
-            trial_id: String. Unique id for this trial.
-            score: Scalar. Lower is better.
-        """
-        pass
+        if self.max_trials and len(self.trials.items()) >= self.max_trials:
+            status = trial_lib.TrialStatus.STOPPED
+            values = None
+        else:
+            response = self.populate_space(trial_id)
+            status = response['status']
+            values = response['values'] if 'values' in response else None
+
+        hyperparameters = self.hyperparameters.copy()
+        hyperparameters.values = values
+        trial = trial_lib.Trial(
+            hyperparameters=hyperparameters,
+            trial_id=trial_id,
+            status=status)
 
-    def report_status(self, trial_id, status, score=None, t=None):
-        """Used by a worker to report the current status of a trial.
+        if status == trial_lib.TrialStatus.RUNNING:
+            self.ongoing_trials[tuner_id] = trial
+            self.trials[trial_id] = trial
+
+        return trial
+
+    def update_trial(self, trial_id, metrics, t=0):
+        """Used by a worker to report the status of a trial.
 
         Args:
             trial_id: A previously seen trial id.
-            status: String, one of "RUNNING", "CANCELLED". A status of
-                "CANCELLED" means a trial has crashed or been deemed
-                infeasible.
-            score: (Optional) Float. The current, intermediate value of the
-                trial's objective.
-            t: (Optional) Float. The current value in a timeseries representing
-                the state of the trial. This is the value that `score` will be
-                associated with.
+            metrics: Dict of float. The current value of this
+                trial's metrics.
+            t: (Optional) Float. Used to report intermediate results. The
+                current value in a timeseries representing the state of the
+                trial. This is the value that `metrics` will be associated with.
 
         Returns:
-            `OracleResponse.STOP` if the trial should be stopped, otherwise 
-            `OracleResponse.OK`.
+            Trial object. Trial.status will be set to "STOPPED" if the Trial
+            should be stopped early.
         """
-        return OracleResponse.OK
+        trial = self.trials[trial_id]
+        self._check_objective_found(metrics)
+        for metric_name, metric_value in metrics.items():
+            trial.metrics.update(metric_name, metric_value, t=t)
+        # To handle early stopping, set Trial.status to "STOPPED".
+        return trial
```

Comment:
Should probably just return the tria status

Reply:
Done